### PR TITLE
fix(core): enable details for analyzer

### PR DESCRIPTION
**Pull request by enkot**
https://github.com/unocss/unocss/issues/2682

Currently Analyzer shows incorrect info, because `config.details` is disabled by default.

### CODE REVIEW

**Comment by antfu** [comment]
I don't actually mind turning `details` on by default on dev mode. This PR's implementation feels a bit verbose and duplicated

**Comment by enkot** [comment]
But in this case user will be able to break the Analyzer by setting "details: false" in config.

File: packages/core/src/config.ts [hunk 1]
```diff
@@ -157,6 +158,7 @@ export function resolveConfig<Theme extends object = object>(
     extractors,
     safelist: getMerged('safelist'),
     separators,
+    details: config.details ?? process.env.NODE_ENV === 'development',
```
Comment (antfu) [inline packages/core/src/config.ts:161]: The core is env agnostic, we shouldn't rely on process. Might be better to rely on envMode and move this logics to the vite/webpack plugin

**Comment by enkot** [comment]
@antfu 3rd attempt :)

**Review by antfu — APPROVED** [review]

**Pull request merged**